universum/configuration_support.py

```diff
@@ -7,15 +8,131 @@
 __all__ = [
     "Step",
     "Configuration",
-    "Variations",
     "set_project_root",
     "get_project_root"
 ]
 
 
 class Step:
     """
-    Step is a collection of configurable project-specific data needed for Universum operations.
+    Step is a collection of configurable project-specific data entries needed for Universum functionality and passed to
+    a single Universum launch. Individual build steps are collected in a :class:`Configuration` object, which supports
+    operations for combination and chaining.
+
+    **Step keys**
+
+    Parameters supplied via named attributes during construction are annotated and can be type-checked for safety.
+    Users may also define custom parameters in `kwargs` - these parameters will be stored internally in `dict` fields
```

Comment:
in a separate `dict` field (as I understand, there's only one). Also, maybe add the field name?

Reply:
each kwarg passed get its own dict field